crates/context/config/src/client/env/config/mutate.rs

```diff
@@ -126,7 +128,37 @@ impl<'a> Method<Starknet> for Mutate<'a> {
     }
 }
 
-impl<'a, T: Transport + Debug> ContextConfigMutateRequest<'a, T> {
+impl<'a> Method<Icp> for Mutate<'a> {
+    type Returns = ();
+
+    const METHOD: &'static str = "mutate";
+
+    fn encode(self) -> eyre::Result<Vec<u8>> {
+        let signer_sk = SigningKey::from_bytes(&self.signing_key);
+
+        let request = ICPRequest::new(signer_sk.verifying_key().rt()?, self.kind.into());
+
+        let signed = ICPSigned::new(request, |b| signer_sk.sign(b))?;
+
+        let encoded = candid::encode_one(&signed)?;
+
+        Ok(encoded)
+    }
+
+    fn decode(response: Vec<u8>) -> eyre::Result<Self::Returns> {
+        if response.is_empty() {
+            // Empty response means success
+            Ok(())
+        } else {
+            // Non-empty response means there was an error message
+            let error_msg = String::from_utf8(response)
+                .map_err(|e| eyre::eyre!("Invalid UTF-8 in error message: {}", e))?;
+            eyre::bail!("{}", error_msg)
+        }
```

Comment:
the error message here would just be the response, without any indication on why it's wrong.

Reply:
Updated